app/controllers/registrations_controller.rb

```diff
@@ -1,40 +1,50 @@
 class RegistrationsController < ApplicationController 
-    def index
-    end
 
-    def sighnin
-        
-    end
+  def index
+  end
 
-    def validateuser
-        user = User.find_by(email: params[:email])
-        if user.present? && user.authenticate(params[:password])
-            session[:user_id]=user.id
-            #redirect_to user_blogs_path @user
-            redirect_to blogs_path
-        else
-            flash[:alert]="Invalid Email or Password"
-            render :sighnin
-        end
-    end
-    def new
-        @user = User.new
-    end
+  def sighnin        
+  end
 
-    def create
-        @user = User.new(user_params)
-        if @user.save
-            session[:user_id]=@user.id
-            #redirect_to user_blogs_path @user
-            redirect_to blogs_path
-        else
-            render :new
-        end
+  def validateuser
+    user = User.where(email: params[:email])
+
+    if user
+      user = user.first 
```

Comment:
this is confusing, use proper naming in all places
when considering names, think whether it is singular or plural
in function names also